pkg/test/suite/run.go

```diff
@@ -0,0 +1,150 @@
+package suite
+
+import (
+	"fmt"
+	"github.com/applike/gosoline/pkg/clock"
+	"github.com/applike/gosoline/pkg/stream"
+	"github.com/applike/gosoline/pkg/test/env"
+	"github.com/stretchr/testify/assert"
+	"reflect"
+	"regexp"
+	"testing"
+)
+
+type testCaseMatcher func(method reflect.Method) bool
+type testCaseBuilder func(suite TestingSuite, method reflect.Method) (testCaseRunner, error)
+type testCaseRunner func(t *testing.T, suite TestingSuite, suiteOptions *suiteOptions, environment *env.Environment)
+
+type testCaseDefinition struct {
+	matcher testCaseMatcher
+	builder testCaseBuilder
+}
+
+var testCaseDefinitions = map[string]testCaseDefinition{}
+
+func Run(t *testing.T, suite TestingSuite, extraOptions ...Option) {
+	var err error
+	var testCases map[string]testCaseRunner
+	var suiteOptions = suiteApplyOptions(suite, extraOptions)
+
+	if testCases, err = suiteFindTestCases(t, suite); err != nil {
+		assert.FailNow(t, err.Error())
+		return
+	}
+
+	if len(testCases) == 0 {
+		return
+	}
+
+	if suiteOptions.envIsShared {
+		runTestCaseWithSharedEnvironment(t, suite, suiteOptions, testCases)
+	} else {
+		runTestCaseWithIsolatedEnvironment(t, suite, testCases, suiteOptions)
+	}
+}
+
+func suiteFindTestCases(t *testing.T, suite TestingSuite) (map[string]testCaseRunner, error) {
+	var err error
+	var regex *regexp.Regexp
+	var testCases = make(map[string]testCaseRunner)
+	var methodFinder = reflect.TypeOf(suite)
+
+	if regex, err = regexp.Compile("^Test"); err != nil {
```

Comment:
No need for a regex when this is just `strings.HasPrefix`

Reply:
already dropped